packages/chain-bridge/src/adapters/StargateAdapter.sol

```diff
@@ -818,6 +1143,137 @@ contract StargateAdapter is Ownable, IBridgeAdapter, ILayerZeroComposer {
         );
     }
 
+    /**
+     * @dev Try to deposit assets to FleetCommander without reverting on failure
+     */
+    function _tryDepositToFleetCommander(
+        address fleetCommander,
+        address asset,
+        uint256 amount,
+        address shareRecipient,
+        bytes memory referralCode,
+        bytes32 operationId,
+        address, // originalUser - not used in current implementation
+        address, // sourceAsset - not used in current implementation
+        uint16 sourceChainId
+    ) internal returns (bool success) {
+        // Validate FleetCommander supports the asset
+        try IFleetCommanderMinimal(fleetCommander).asset() returns (
```

Comment:
fleet commanders are listed in the harbor command contract.

from Admirals Quarters:

```ts
    function _validateFleetCommander(address fleetCommander) internal view {
        if (
            !IHarborCommand(harborCommand()).activeFleetCommanders(
                fleetCommander
            )
        ) {
            revert InvalidFleetCommander();
        }
    }
```

Reply:
Done